### ISSUE TITLE
The retro achievement password cannot contain $ or #

### ISSUE THREAD

**Issue by brunodupim08** [batocera-linux/batocera-emulationstation#1423]
I can't say if the error is from batocera or retroarch, but all I found is that when using some characters in the retro achievement login password, an error prevents me from logging in.

**Comment by lbrpdx** [comment]
Did you try to escape your special characters with a backslash like explained in the wiki?https://wiki.batocera.org/retroachievements_settings#manually_adjusting_retroachievement_settings

**Comment by brunodupim08** [comment]
yes, but the problem still occurred, my password was 32 characters with many symbols.
The best solution was to remove them all and replace them with numbers and letters, but this should be informed in the interface itself, I found out about this problem by opening batocera.conf.

**Comment by brunodupim08** [comment]
my doubt is, the problem of needing \ is in retroarch or in batocera? if it's something bash related maybe i can look into it.

**Comment by n2qz** [comment]
For the most part, ES is handling the password correctly.

- The comments in the wiki are wrong, at least with respect to retroarch with cheevos.  Nothing that processes the passwords or other cheevos configuration items handles backslash characters as an escape.
- Storing and retrieving the password credentials in batocera.conf seems fine, the way ES is handling it today
- configgen seems fine with the password handling as well, even when special characters are present
- Retroarch has several config parsing bugs with these credentials.  "#" is always handled as a comment character if it appears in an unquoted configuration value.  Quoting the value when handing it from configgen to retroarch solves that, but then there is no way to include a double-quote '"' character in the password because the double-quote always ends the configuration value.
- Most configgens are using the token, but in addition to retroarch there are a few for standalone emulators that still use the password, and these will need validation as well

The best approach seems to be to switch from using the password to using the cheevos token.  Several configgens already do this, and it works fine with retroarch as well.

The downside to this approach is the cheevos token expires after 14 days.  My intention is to add code to the ES startup to log in to retroachivements with the stored username/password and generate a new token.  It may also be necessary to set a timer to generate a new token occasionally, in case ES is left running for weeks.

**Comment by n2qz** [comment]
Transferring this to batocera.linux as ES will get a small related feature to ensure cheevos tokens are refreshed often, but most of the fixes/workarounds will be on the Batocera side.
